### ISSUE TITLE
Assets with normalized characters duplicated on Upload

### ISSUE THREAD

**Issue by jstoks** [statamic/eloquent-driver#251]
Assets with filenames that include characters that will be replaced by sanitization will cause a duplicate record in the `assets_meta` table. When viewing the index of the assets container there will be a row for the original filename with 0 size and timestamp.

I tested this using the local file system and S3. They both have this issue.

Additionally, this phantom file causes the asset index to slow down to a crawl (25 seconds) when using S3. Deleting the erroneous assets_meta record fixes the slowdown. 

### Example

In this example I'm uploading a file with spaces in the filename.

![Screenshot 2024-03-06 at 6 05 06 PM](https://github.com/statamic/eloquent-driver/assets/2073157/6c45478f-aada-479d-ae2a-667063a6ea80)

```
-[ RECORD 1 ]-------
id         | 27
container  | image_assets
folder     | /
basename   | 2020 07 26.jpeg
filename   | 2020 07 26
extension  | jpeg      
path       | 2020 07 26.jpeg
meta       | {"data": []}
created_at | 2024-03-07 00:16:02
updated_at | 2024-03-07 00:16:02
-[ RECORD 2 ]-------
id         | 28
container  | image_assets
folder     | /
basename   | 2020-07-26.jpeg
filename   | 2020-07-26
extension  | jpeg      
path       | 2020-07-26.jpeg
meta       | {"data": [], "size": 149857, "width": 680, "height": 510, "duration": null, "mime_type": "image/jpeg", "last_modified": 1709770562}
created_at | 2024-03-07 00:16:02
updated_at | 2024-03-07 00:16:02
```


### Support Details

```
Environment                                                                                                                                     
Application Name: Laravel                                                                                                                       
Laravel Version: 10.46.0                                                                                                                        
PHP Version: 8.1.27                                                                                                                             
Composer Version: 2.7.1                                                                                                                         
Environment: local                                                                                                                              
Debug Mode: ENABLED                                                                                                                             
URL: localhost                                                                                                                                  
Maintenance Mode: OFF                                                                                                                           
                                                                                                                                                
Cache                                                                                                                                           
Config: NOT CACHED                                                                                                                              
Events: NOT CACHED                                                                                                                              
Routes: NOT CACHED
Views: NOT CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: pgsql
Logs: cli_stack / background, bugsnag
Mail: smtp
Queue: sync
Scout: database_index
Session: file

Statamic
Addons: 1
Antlers: runtime
Sites: 1
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.52.0 PRO

Statamic Addons
statamic/eloquent-driver: 3.2.0

Statamic Eloquent Driver
Asset Containers: eloquent
Assets: eloquent
Blueprints: eloquent
Collection Trees: eloquent
Collections: eloquent
Entries: eloquent
Forms: eloquent
Global Sets: eloquent
Global Variables: eloquent
Navigation Trees: eloquent
Navigations: eloquent
Revisions: eloquent
Taxonomies: eloquent
Terms: eloquent
```

**Comment by ryanmitchell** [comment]
Can you see if this fixes it: https://github.com/statamic/eloquent-driver/pull/252 ?

**Comment by jstoks** [comment]
> Can you see if this fixes it: #252 ?

This does fix it. I tested it against filesystem and S3. Thank you for the fast turn around.
